main.cpp

```diff
@@ -183,123 +191,171 @@ int main()
 
 struct FloatType
 {
-    float add(float lhs, float rhs)
+    FloatType(float) : value(new float) {}
```

Comment:
This does not initialize `value` with the constructor argument.
what you want is:
```
FloatType( float f ) : value( new float(f) ) { }
```
- [ ] revise